src/java.desktop/share/classes/javax/swing/plaf/basic/BasicSliderUI.java

```diff
@@ -1581,7 +1615,7 @@ else if ( slider.getOrientation() == JSlider.HORIZONTAL ) {
                   g.drawLine(w-1, 1, w-1,  h-2 );          // right
             }
         }
-
+        g.setClip(clip);
```

Comment:
Since it is normally seen in Metal L&F, shouldn't we do the clip setting in MetalSliderUI.paintThumb() instead?
If we are doing it in BasicSliderUI, can you please explain why are we not seeing it in other L&Fs like nimbus or motif?

Reply:
The bug in the motif was fixed by the JDK-8032219, the code in the BasicSliderUI is never used by the Metal L&F, I tested that code by the commented out the overridden code in the MetalSliderUI.

Reply:
If the code in BasicSliderUI is not used by Metal L&F, then, all the more reason to have the clip setting in MetalSliderUI.paintThumb(), no?
What about nimbus? Why it is not seen in nimbus l&f? 
I guess only reason I can think of putting the code in BasicL&F is if it's a generic issue and it will solve multiple L&F but it seems the issue only affects MetalL&F.

Reply:
I have set the clip in the methods which do not draw the thumb properly - draw it outside the bounds of the thumb(it is a limitationGraphics). Only three places were affected, the motif - fixed in JDK-8032219, the old default metal L&F(steel) fixed in this fix in the MetalIconFactory.java. And the code in the BasicSliderUI is not executed by the metal but could be triggered by commenting on some code.